commands/usernote.js

```diff
@@ -95,31 +95,22 @@ module.exports = {
           }
         });
         if (notes) {
-          if(!(args.all === false)) {
-              return Promise.map(notes.ns, note => formatNote(note, args.subreddit, channel))
-                .map((note, index) => `(${index}) ${note}`)
-                .then(formattedNotes => [`Notes on /u/${args.user} on /r/${args.subreddit}:`].concat(formattedNotes));
-          } else {
-              return Promise.map(notes.ns, note => formatNote(note, args.subreddit, channel))
-                .map((note, index) => { 
-                  if(index<=4) {
-                    return `(${index}) ${note}`;
-                  }
-                })
-                .then(formattedNotes => [`Notes on /u/${args.user} on /r/${args.subreddit}:`].concat(formattedNotes))
-                .then(array => {
-                  var notesRemaining = _.size(notes.ns) >= 5 ? _.size(notes.ns) - 5 : _.size(notes.ns);
-                  return array.concat([`... and ${notesRemaining} more usernotes. Use \`.tag /u/username --all\` to display all.`]);
-                });
-          }
-          
+          return Promise.map(
+            args.all !== false ? notes.ns : notes.ns.slice(0,5),
+            note => formatNote(note, args.subreddit, channel)
+          )
+            .map((note, index) => `(${index}) ${note}`)
+            .then(formattedNotes => [`Notes on /u/${args.user} on /r/${args.subreddit}:`].concat(formattedNotes))
+            .then(arrayOfNotes => {
+              return _.size(notes.ns) >= 5 ? arrayOfNotes.concat([`... and ${_.size(notes.ns) - 5} more usernotes. Use \`.tag /u/username --all\` to display all.`]) : arrayOfNotes;
```

Comment:
I think this is going to display the "...more usernotes" message even when the `--all` flag is used. It might be better to do something like this:

```js
// ... stuff from before
  .map((note, index) => `(${index}) ${note}`)
  .then(formattedNotes =>
    formattedNotes.length < notes.ns.length
      ? formattedNotes.concat(`...and ${notes.ns.length - formattedNotes.length} more (use --all to display)`)
      : formattedNotes
  )
  .then(formattedNotes => [`Notes on /u/${args.user} on /r/${args.subreddit}:`].concat(formattedNotes))
```

This also avoids repeating the number 5 in multiple places, which will make it easier to change if we decide we want to set a different cutoff in the future.